server/src/test/java/org/candlepin/dto/api/v1/ComplianceStatusDTOTest.java

```diff
@@ -0,0 +1,134 @@
+            reason.setKey("test-key-" + i);
+            reason.setMessage("test-msg-" + i);
+
+            Map<String, String> attributes = new HashMap<>();
+            attributes.put("a1-" + i, "v1-" + i);
+            attributes.put("a2-" + i, "v2-" + i);
+            attributes.put("a3-" + i, "v3-" + i);
+
+            reasons.add(reason);
+        }
+
+        Map<String, DateRange> ranges = new HashMap<>();
+
+        for (int i = 0; i < 3; ++i) {
+            DateRange range = new DateRange();
+
+            Calendar sdc = Calendar.getInstance();
+            sdc.add(Calendar.HOUR, i);
+
+            range.setStartDate(sdc.getTime());
+
+            Calendar edc = Calendar.getInstance();
+            edc.add(Calendar.HOUR, i);
+            edc.add(Calendar.YEAR, 1);
+
+            range.setEndDate(edc.getTime());
+
+            ranges.put("test_prod-" + i, range);
+        }
+
+        Map<String, Set<EntitlementDTO>> compliantProducts = new HashMap<>();
+        Map<String, Set<EntitlementDTO>> partiallyCompliantProducts = new HashMap<>();
+        Map<String, Set<EntitlementDTO>> partialStacks = new HashMap<>();
+
+        for (int i = 0; i < 3; ++i) {
+            compliantProducts.put("p" + i, this.generateEntitlementDTOs(3));
+            partiallyCompliantProducts.put("p" + (3 + i), this.generateEntitlementDTOs(3));
+            partialStacks.put("s" + i, this.generateEntitlementDTOs(3));
+        }
+
+        this.values = new HashMap<>();
+        this.values.put("Status", "test-status");
+        this.values.put("Compliant", Boolean.TRUE);
+        this.values.put("Date", new Date());
+        this.values.put("CompliantUntil", new Date());
+
+        this.values.put("CompliantProducts", compliantProducts);
+        this.values.put("NonCompliantProducts", Arrays.asList("p1", "p2", "p3"));
+        this.values.put("PartiallyCompliantProducts", partiallyCompliantProducts);
+        this.values.put("PartialStacks", partialStacks);
+        this.values.put("ProductComplianceDateRanges", ranges);
+        this.values.put("ComplianceReasons", reasons);
+    }
+
+    private Set<EntitlementDTO> generateEntitlementDTOs(int count) {
+        Set<EntitlementDTO> ents = new HashSet<>();
+
+        for (int i = 0; i < count; ++i) {
+            EntitlementDTO dto = new EntitlementDTO();
+            dto.setId("test-dto-" + i);
+
+            ents.add(dto);
+        }
+
+        return ents;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Object getInputValueForMutator(String field) {
+        return this.values.get(field);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Object getOutputValueForAccessor(String field, Object input) {
+        // Since we're returning primitives for two methods, we need to convert them as necessary
+        if (field != null && (field.equals("Result") || field.equals("Standalone")) && input == null) {
+            return false;
+        }
```

Comment:
Same as the other test class, some leftovers here

Reply:
Fixed.